core/src/operations/decorators/asmop_info.rs

```diff
@@ -0,0 +1,37 @@
+use crate::utils::string::String;
+
+// ASMOP INFO
+// ================================================================================================
+
+/// Contains information corresponsing to an assembly instruction (only applicable in debug mode)
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AsmOpInfo {
+    op: String,
+    cycles: u8,
```

Comment:
minor nit: would prefer `num_cycles` here and also `get_num_cycles()` and `set_num_cycles()` in the implementation.

Reply:
Fixed